backend/cfg/cfgize.ml

```diff
@@ -596,11 +586,10 @@ module Trap_depth_and_exns = struct
 
   type handler_table = handler_stack Label.Tbl.t
 
-  let record_handler :
-      type a.
-      handler_stack -> handler_table -> can_raise:(a -> bool) -> a -> unit =
-   fun stack table ~can_raise instr ->
-    if can_raise instr
+  let record_handler : handler_stack -> handler_table -> can_raise:bool -> unit
```

Comment:
What is the rationale behind that change?

Reply:
I thought it was simpler to compute the boolean value of `can_raise` at the couple of call sites instead of passing the function and the instruction, because the instruction is not used for anything else.

Reply:
I guess it is in the eye of the beholder. I have nothing
against the new code and am fine with having it part of
this pull request, but we should maybe do such
refactorings is dedicated pull requests so that the reviewer
does not have to double check whether it is semantics
preserving or part of larger changes.